execution/step_invariant/step_invariant.go

```diff
@@ -142,7 +142,7 @@ func (u *stepInvariantOperator) cacheInputVector(ctx context.Context) error {
 		}
 		defer u.next.GetPool().PutVectors(in)
 
-		if len(in) == 0 || len(in[0].Samples) == 0 {
+		if len(in) == 0 || (len(in[0].Samples) == 0 && len(in[0].Histograms) == 0) {
```

Comment:
A bit of a bundled bugfix here; we can do that in its own PR i think too; wdyt?

Reply:
I am ok to fix it this pr